### PR TITLE
fix(dark-factory): stop --implement noise on empty In Progress column

### LINKED ISSUE #414: factory --implement: jq error and #null warnings during board scan
## What

Every `factory-agent.sh --implement` tick logs two duplicate warnings to stdout and two `jq` errors to stderr, even when there are zero In Progress items on the board:

```
[20:04:26] --implement (Phase A): 0 In Progress item(s)
[20:04:27] WARNING: fetch_issue_comments failed for #null; skipping
[20:04:27] WARNING: fetch_issue_comments failed for #null; skipping
```

```
jq: error (at <stdin>:0): Cannot index string with string "id"
jq: error (at <stdin>:0): Cannot index string with string "id"
```

## Where

Surfaced on the Mac mini after Wave 4 launchd install, on the first live tick (`launchd-factory-stdout.log` / `launchd-factory-stderr.log`). Code likely lives in `scripts/lib/factory-project.mjs` and/or `scripts/lib/factory-bundle.mjs`.

## Theory

`--implement` reports 0 In Progress items but still attempts `fetch_issue_comments` for two phantom `#null` entries. The `jq: Cannot index string with string "id"` suggests jq is being fed a string where an object/array was expected — probably a board item with non-standard `content` shape (e.g. a draft issue, an archived item, or an item whose linked issue has been deleted) is leaking into a downstream `.content.id` lookup.

## Impact

- No functional impact at Phase A — `--implement` still no-ops correctly and the tick exits 0.
- Will spam stderr every 5 min indefinitely. Risk of masking real `jq` errors later.

## Acceptance

- `--implement` produces no `#null` warnings and no `jq` errors in stderr on a clean run.
- A regression test guards the parse against the same shape (e.g. a board item with `content == null`).

## Affected platforms

- [ ] Web
- [ ] iOS
- [ ] Android
- [ ] macOS

Internal tooling — `parity:web-only` not applicable; this is `scripts/` and runs on the Mac mini only.

### PR DESCRIPTION
**Pull request by JakubAnderwald**
## Summary

- `scripts/factory-agent.sh --implement` was emitting two `WARNING: fetch_issue_comments failed for #null` lines and two `jq: Cannot index string with string "id"` errors per 5-minute launchd tick whenever the kanban board had zero "In Progress" cards. Root cause is the BSD-`seq` gotcha: `seq 0 -1` (from `INPROG_COUNT - 1` when count is 0) auto-flips to a descending sequence on macOS and prints `0\n-1`, so the loop ran twice with bogus indices and fetched comments for issue number "null".
- Adds an early-return guard mirroring the existing `--plan` block, and switches both `seq`-based loops in the script to C-style `for ((IDX=0; IDX<COUNT; IDX++))` so the zero-count case is structurally safe regardless of seq flavor.
- Adds a small node-test regression that locks in the new loop form and documents the old BSD-seq behavior for future readers.

Closes #414

## Test plan

- [x] `node --test scripts/__tests__/factory-agent-implement.test.mjs` — all 3 cases pass on darwin
- [x] `node --test scripts/__tests__/*.test.mjs` — full scripts suite (278 tests) still green
- [x] `bash -n scripts/factory-agent.sh` — syntax-clean
- [x] `pnpm format:check`, `pnpm lint`, `pnpm typecheck` — all green (warnings unchanged from main)
- [ ] After merge, tail `~/Library/Logs/eu.drafto.factory/launchd-factory-stderr.log` on the Mac mini for one tick (~5 min) and confirm no `#null` / `jq` errors

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved edge-case handling for zero-item iterations in the factory agent script.

* **Tests**
  * Added regression test suite to verify correct behavior of iteration logic, including zero-item edge cases and platform-specific scenarios.

[Review Change Stack](https://app.coderabbit.ai/change-stack/JakubAnderwald/drafto/pull/419?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: scripts/__tests__/factory-agent-implement.test.mjs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    if (platform === "darwin") {
+      assert.deepEqual(
+        lines,
+        ["0", "-1"],
+        "BSD seq on darwin prints 0 and -1 for `seq 0 -1` — this is what caused #414",
+      );
+    }
+    // On linux GNU seq, lines === []; no assertion needed.
+  });
+});
```
[the remark this answers is not in the document]
Comment (JakubAnderwald) [reply scripts/__tests__/factory-agent-implement.test.mjs:80]: Fixed in d9d4e6f. Added a structural assertion that reads `scripts/factory-agent.sh` and verifies both `--implement` and `--plan` loops use the C-style form, the zero-count guard is in place, and the `seq`-based form cannot reappear without breaking the test.

[the remark this answers is not in the document]
Comment (JakubAnderwald) [reply scripts/__tests__/factory-agent-implement.test.mjs:80]: Thanks for confirming. Resolving the thread.
